src/handle_join_leave.cxx

```diff
@@ -202,12 +202,14 @@ void raft_server::sync_log_to_new_srv(ulong start_idx) {
     p_db("[SYNC LOG] peer %d start idx %llu, my log start idx %llu\n",
          srv_to_join_->get_id(), start_idx, log_store_->start_index());
     // only sync committed logs
-    int32 gap = (int64_t)quick_commit_index_ - (int64_t)start_idx;
+    ulong gap = ( quick_commit_index_ > start_idx )
+                ? ( quick_commit_index_ - start_idx )
+                : 0;
     ptr<raft_params> params = ctx_->get_params();
     if ( gap < params->log_sync_stop_gap_ ||
```

Comment:
Now the compiler complains about the comparison between signed/unsigned integers. Let me fix it.